app/providers/utils.py

```diff
@@ -0,0 +1,47 @@
+"""Utils for logging retry information."""
+
+from loguru import logger
+from tenacity import RetryCallState
+
+
+def log_on_retry(retry_state: RetryCallState) -> None:
+    """Log the result of the last call attempt if it failed.
+
+    Args:
+        retry_state (RetryCallState): The state of the last call attempt.
+
+    """
+    fn_name = f'{retry_state.fn.__module__}.{retry_state.fn.__name__}'
+
+    logger.info(
+        'Retrying {}: attempt {} raised {}: "{}" retrying in {:.3f} seconds.',
+        fn_name,
+        retry_state.attempt_number,
+        retry_state.outcome.exception().__class__.__name__,
+        retry_state.outcome.exception(),
+        retry_state.next_action.sleep,
+    )
```

Comment:
mypy doesn't like anything that's going on here. Will verify with @k-macmillan what we want to do about that.
[failed linting](https://github.com/department-of-veterans-affairs/va-enp-api/actions/runs/12245671790/job/34160033994?pr=94)

Reply:
Spoke with Kyle, we will use `type: ignore` since it's a third party library and we can't control how they implement their code.